test/integration/framework/apiserver.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019, Verizon Media Inc.
+// Licensed under the terms of the 3-Clause BSD license. See LICENSE file in
+// github.com/yahoo/k8s-athenz-istio-auth for terms.
+
+// +build integration
+
+package framework
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/coreos/etcd/embed"
+	"github.com/yahoo/k8s-athenz-istio-auth/test/integration/fixtures"
+	athenzdomainclientset "github.com/yahoo/k8s-athenz-syncer/pkg/client/clientset/versioned"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+)
+
+type Framework struct {
+	AthenzDomainClientset *athenzdomainclientset.Clientset
+}
+
+// runEtcd will setup up the etcd configuration and run the etcd server
+func runEtcd() error {
+	etcdDataDir, err := ioutil.TempDir(os.TempDir(), "integration_test_etcd_data")
+	if err != nil {
+		return err
+	}
+
+	config := embed.NewConfig()
+	config.Dir = etcdDataDir
+	_, err = embed.StartEtcd(config)
+	return err
+}
+
+// runApiServer will setup the api configuration and run the api server
+func runApiServer() (*rest.Config, error) {
+	s := options.NewServerRunOptions()
+
+	listener, err := net.Listen("tcp4", "127.0.0.1:9999")
```

Comment:
we can replace s.InsecureServing.Listener with BindPort and BindNetwork to be consistent with BindAddress
```
s.InsecureServing.BindAddress = net.ParseIP("127.0.0.1")
s.InsecureServing.BindPort = 9999
```